source/Playnite/GamesEditor.cs

```diff
@@ -467,6 +467,206 @@ public void OpenGameLocation(Game game)
+                                }
+
+                                if (AppSettings.InstallSizeScanUseSizeOnDisk)
+                                {
+                                    size += FileSystem.GetFileSizeOnDisk(file);
+                                }
+                                else
+                                {
+                                    size += FileSystem.GetFileSize(file);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            if (AppSettings.InstallSizeScanUseSizeOnDisk)
+                            {
+                                size += FileSystem.GetFileSizeOnDisk(rom.Path);
+                            }
+                            else
+                            {
+                                size += FileSystem.GetFileSize(rom.Path);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    if (game.InstallDirectory.IsNullOrEmpty())
+                    {
+                        return;
+                    }
+
+                    var expandedGame = game.ExpandGame();
+                    if (!FileSystem.DirectoryExists(expandedGame.InstallDirectory))
+                    {
+                        return;
+                    }
+
+                    if (checkLastScanDate)
+                    {
+                        var addedAfterLastCheck = expandedGame.LastSizeScanDate == null || expandedGame.Added != null && expandedGame.Added > expandedGame.LastSizeScanDate;
+                        if (!addedAfterLastCheck && expandedGame.LastSizeScanDate != null && FileSystem.DirectoryGetLastWriteTime(expandedGame.InstallDirectory) < expandedGame.LastSizeScanDate)
+                        {
+                            return;
+                        }
+                    }
+
+                    if (AppSettings.InstallSizeScanUseSizeOnDisk)
+                    {
+                        size = FileSystem.GetDirectorySizeOnDisk(expandedGame.InstallDirectory);
+                    }
+                    else
+                    {
+                        size = FileSystem.GetDirectorySize(expandedGame.InstallDirectory);
+                    }
+                }
+
+                game.LastSizeScanDate = DateTime.Now;
+                var ulongSize = (ulong)size;
+                if (game.InstallSize != ulongSize)
+                {
+                    game.InstallSize = ulongSize;
+                }
+
+                if (updateGame)
+                {
+                    Database.Games.Update(game);
+                }
+            }
+            catch (Exception e) when (!PlayniteEnvironment.ThrowAllErrors)
+            {
+                logger.Error(e, $"Failed to get InstallSize from game {game.Name} with install dir {game.InstallDirectory}");

```

Comment:
This should not show an error message here. It could lead to error message spam during scan of multiple games.

It should imo just log error and throw and then let the called (library or manual scan) to deal with how to show error. Meaning that during library scan it should just silently fail and during manual recalculation you should show error dialog. But ideally not a separate error dialog for every game, that could get messy.

Reply:
Done in https://github.com/JosefNemec/Playnite/pull/3017/commits/2b0e2f8be11add23270c5cc2d6c268fd3bee7d89